bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productcarousel/ProductCarouselImpl.java

```diff
@@ -91,94 +113,67 @@
     private Page productPage;
     private List<String> baseProductSkus = Collections.emptyList();
 
-    private AbstractProductsRetriever productsRetriever;
+    private ProductsRetriever productsRetriever;
 
     @PostConstruct
     private void initModel() {
+        if (StringUtils.isBlank(selectionType)) {
+            selectionType = PRODUCT_SELECTION;
+        }
+
+        productCount = resource.getValueMap().get(PRODUCT_COUNT_PROPERTY, currentStyle.get(PRODUCT_COUNT_PROPERTY, DEFAULT_PRODUCT_COUNT));
+        productCount = Math.min(Math.max(MIN_PRODUCT_COUNT, productCount), MAX_PRODUCT_COUNT);
```

Comment:
I don't understand what the added value of a max count would be, in any way you should cover those conditions with unit tests.

Reply:
The max is protection for large any possible large value coming from the JCR property.